src/placeos-edge/transport.cr

```diff
@@ -80,8 +81,17 @@ module PlaceOS::Edge
         socket_lock.synchronize do
           begin
             socket?.try(&.ping)
+            @ping_failures = 0
```

Comment:
Oh I thought these success lines would go into the else block of a rescue? Or does it make more sense in the begin?

Reply:
this function basically runs in a loop, if a ping succeeds we reset the count
if there is a failure then the count is incremented.

if it fails for 5min (default interval) then we exit the process, which will be restarted by the service manager.
basically a failsafe if it ever gets into a bad way.

note it'll this will only ever occur if there is a connection failure and then edge fails to reconnect

Reply:
basically ensures we never have to call anyone if something goes wrong
it'll always recover